### ISSUE TITLE
Search - Own email address is displayed twice on /search/filters page

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#47537]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.21-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N (Can't check - new feature)
Issue was found when executing this PR: https://github.com/Expensify/App/pull/46962
**Email or phone of affected tester (no customers):** gocemate+a980@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to /search/filters
3. Click on From filter and check email list
4. Click on To filter and check email list

## Expected Result:
Self email address should be displayed once

## Actual Result:
Self email address is displayed twice, both From and To filters

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/eb2e6077-45f9-4ea7-96cc-a4f8218a2aea

![image (233)](https://github.com/user-attachments/assets/8bf30739-433b-4075-8c1f-76354c10ef29)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #vip-vsb

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There are 2 current user options in the search From and To filters.

### What is the root cause of that problem?
In the search filter From/To, we put the current user option at the top of the list and then the recent reports.
https://github.com/Expensify/App/blob/5e5bff3630958264c22a34f083879792ff8ee6ab/src/components/Search/SearchFiltersParticipantsSelector.tsx#L110-L122

However, the recent reports also contains a self DM, so the list contains 2 current user option. That's because we include self DM when getting the options (the last param).
https://github.com/Expensify/App/blob/5e5bff3630958264c22a34f083879792ff8ee6ab/src/components/Search/SearchFiltersParticipantsSelector.tsx#L59-L80

### What changes do you think we should make in order to solve the problem?

We shouldn't include the self DM into the recent report because we already put the current user at the top of the list. We can set it to false or just remove the param.

**Comment by techievivek** [comment]
@bernhardoj can you please link the GH that caused this regression?

**Comment by techievivek** [comment]
This seems to be coming from here https://github.com/Expensify/App/pull/46962

**Comment by techievivek** [comment]
I will just inform on the PR and it can be fixed as a followup, and nothing significant to be a blocker.

**Comment by luacmartins** [comment]
I agree this is not a blocker. @Kicu @rayane-djouah since we worked on this PR together.

**Comment by Kicu** [comment]
hey I did the filters and I will take a look today at this issue and try to come up with a fix. Please assign me

**Comment by Kicu** [comment]
PR with a simple fix is ready.

@bernhardoj your suggestion was actually correct and I ended up doing exactly the same thing you suggested - so switching the argument-flag for including selfDM. Feels the cleanest
